pkg/types/validation/installconfig.go

```diff
@@ -699,3 +703,35 @@ func validateFIPSconfig(c *types.InstallConfig) field.ErrorList {
 	}
 	return allErrs
 }
+
+// validateCapabilities checks if additional, optional OpenShift components are specified in the
+// install-config to be included in the installation.
+func validateCapabilities(c *types.Capabilities, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allCapabilitySets := sets.NewString()
+	allAvailableCapabilities := sets.NewString()
+	// Create a map and slice of *all* available sets and capabilities across all capability sets
+	for baselineSet, capabilities := range configv1.ClusterVersionCapabilitySets {
+		if !allCapabilitySets.Has(string(baselineSet)) {
+			allCapabilitySets.Insert(string(baselineSet))
+		}
```

Comment:
It is a set, so the insert is guaranteed not to create a duplicate.
```suggestion
		allCapabilitySets.Insert(string(baselineSet))
```